erroranalysis/erroranalysis/_internal/surrogate_error_tree.py

```diff
@@ -182,20 +254,97 @@ def compute_error_tree(analyzer,
     filtered_indexed_df[DIFF] = diff
     filtered_indexed_df[TRUE_Y] = true_y
     filtered_indexed_df[PRED_Y] = pred_y
-    dumped_model = surrogate._Booster.dump_model()
-    tree_structure = dumped_model["tree_info"][0]['tree_structure']
-    max_split_index = get_max_split_index(tree_structure) + 1
-    cache_subtree_features(tree_structure, dataset_sub_names)
-    tree = traverse(filtered_indexed_df,
-                    tree_structure,
-                    max_split_index,
-                    (categories_reindexed,
-                     cat_ind_reindexed),
-                    [],
-                    dataset_sub_names,
-                    metric=analyzer.metric,
-                    classes=analyzer.classes)
-    return tree
+    return surrogate._Booster, filtered_indexed_df, categorical_info
+
+
+def get_surrogate_booster_pyspark(filtered_df, analyzer, max_depth,
+                                  num_leaves, min_child_samples):
+    """Get surrogate booster for pyspark dataframe.
+
+    Creates the surrogate model trained on errors and returns the booster.
+
+    :param filtered_df: The filtered dataframe.
+    :type filtered_df: pyspark.sql.DataFrame
+    :param analyzer: The error analyzer containing the categorical
+        features and categories for the full dataset.
+    :type analyzer: BaseAnalyzer
+    :param max_depth: The maximum depth of the surrogate tree trained
+        on errors.
+    :type max_depth: int
+    :param num_leaves: The number of leaves of the surrogate tree
+        trained on errors.
+    :type num_leaves: int
+    :param min_child_samples: The minimal number of data required to
+        create one leaf.
+    :type min_child_samples: int
+    :return: The extracted booster from the surrogate model and the
+        scored dataset.
+    :rtype: (Booster, pyspark.sql.DataFrame)
+    """
+    # compute the pred_y column
+    scored_data = analyzer.model.transform(filtered_df.to_spark())
+    diff_data = scored_data.withColumn(
+        DIFF,
+        F.when(F.col(analyzer.true_y) != F.col(PREDICTION),
+               1).otherwise(0))
+    if analyzer.model_task == ModelTask.CLASSIFICATION:
+        diff_data = diff_data.drop(PREDICTION, RAW_PREDICTION, PROBABILITY)
+    else:
+        diff_data = diff_data.drop(PREDICTION)
+    model = create_surrogate_model_pyspark(analyzer, diff_data, max_depth,
+                                           num_leaves, min_child_samples)
+    # TODO: update lightgbm in pyspark to get around file requirement
```

Comment:
Is that a separate PR somewhere?

Reply:
yes I will send a PR in https://github.com/microsoft/SynapseML/ for this at some point